repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngine.java

```diff
@@ -399,6 +411,54 @@ private boolean shouldCache(QueryOptions options, List<Node> nodes, int required
         return frs;
     }
 
+    private void performTmdqSelect(String statement, DBQuery dbQuery, int requiredNodes, ResultHandler<Node> handler)
+    {
+        if (usePagingQuery)
+        {
+            performTmdqSelectPaging(statement, dbQuery, requiredNodes, handler);
+        }
+        else
+        {
+            performTmdqSelectStreaming(statement, dbQuery, handler);
+        }
+    }
+
+    private void performTmdqSelectStreaming(String statement, DBQuery dbQuery, ResultHandler<Node> handler)
+    {
+        template.select(statement, dbQuery, handler);
+    }
+
```

Comment:
Straight forward but a small test that returns 2 to 3 pages of data would be good.

Reply:
Added test cases comparing results at different page sizes to show that paging has np impact on the overall result